### ISSUE TITLE
Use the new Path.join method to append to a path.

### ISSUE THREAD

**Issue by kendrickt** [seattle-uat/civiform#638]
**Is your feature request related to a problem? Please describe.**
There are many instances of path concatenation that looks like `path.toBuilder().append(something).build()`. There is a `Path.join(something)` method that should be used.

**Describe the solution you'd like**
I think the entire `Path.Builder` can be removed. `Path.join` should be used, but may need to be updated because it is using the bulider.

**Additional context**

**Done when**
`Path.Builder` is removed, and `Path.join` is used.

**Comment by davidrhallartificialeducation** [comment]
Look forward to tackling this issue.  I was going to compare the return types of `Path.Builder` and `Path.join`, and then look at each of the `Path.Builder` use cases to make sure they can be replaced

**Comment by kendrickt** [comment]
@davidrhallartificialeducation sounds good. If you find any usages of `Path.Builder` that doesn't look like a `Path.join` link them here. I didn't find any but I only looked for a minute.

**Comment by davidrhallartificialeducation** [comment]
I've replaced the instances of `path.toBuilder().append(XYZ).build()` with `path.join(XYZ`.  The only instances I found were inside of ApplicantServiceImpl.  I've completed the following:

* Replaced `path.toBuilder().append(XYZ).build()` with `path.join(XYZ)`.
* All builder append usages were inside ApplicantServiceImpl
* Added `pathJoin()` unit test to PathTest
* Verified passing unit tests for PathTest & ApplicantServiceImplTest
